transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoHandler.java

```diff
@@ -110,6 +149,40 @@ public int run(IoExecutionContext context) {
         return completionQueue.process(this::handle);
     }
 
+    void submitBeforeIO(Runnable runnable) {
```

Comment:
If we pull in https://github.com/netty/netty/pull/14700 its possible to make this much "faster" by basically running this runnable directly if in the event loop and if not just calling `eventLoop.execute(runnable)`. This way there is no extra syscall when we are already in the event loop thread (which we should be most of the time) and even if not there is no extra queue.

Reply:
That's awesome! I'll make the modification here after this #14700 PR is merged into 4.2.

Reply:
https://github.com/netty/netty/pull/14701 was merged